### ISSUE TITLE
Fix accessibility of the post summary controls

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#63992]
### Description

Follow-up to https://github.com/WordPress/gutenberg/issues/470
See https://github.com/WordPress/gutenberg/issues/63308

All the controls (buttons and links) in the post summary panel are labeled in an unclear and  misleading way as their labels uses the _value_ or _state_ of the control rather than communicating what the control does or is about.
This was originally discussed long time ago in https://github.com/WordPress/gutenberg/issues/470. Since then there have been minor improvements but the underlying issue is still not fully solved.

Buttons should communicate what their action is about.
Links should communicate the link destination or purpose.
They are not places to communicate a value.

Screenshot of the current UI:

![Screenshot 2024-07-11 at 16 34 29](https://github.com/user-attachments/assets/22e9dc88-f1de-490d-919e-e2fcdd61d28d)

In the following screenshot I removed the CSS to better illustrate the buttons and their text labels:

![status publish etc button labels](https://github.com/user-attachments/assets/e63d4e9e-8384-4f57-88f1-9916cc22c723)

All these buttons text is the currently selected value, which is less than ideal.

The only minor improvement since https://github.com/WordPress/gutenberg/issues/470 is that some of these buttons have now an aria-label that gives more context by prepending the button action to the value. Still, less than ideal because the visible text mismatches the actual accessible name. The mismatch is a barrier for sighted screen reader users and speech recognition software users.

Examples of the aria-labels:

- `Change post status: Published`
- `Change date: July 22, 2024 1:40 pm UTC+5:45`
- `Change link: my-post-title`
- `Change author: admin`
- `Template options` (entirely mismatch the visible text)
- `Change discussion options`

Worth reminding the panel may render additional controls depending on the edited object e.g. Format, Parend, Order. It may also render values that arent interactive controls e.g. Sync status.

https://github.com/WordPress/gutenberg/issues/63308 reports additioanl issues for these controls, for example:
- The Revisions link text is just a number with no other context.
- Some of these controls use tooltips to add more context but they do that in an inconsistent way.
- There's no semantic indication that some of these controls open the related popovers.

While the addition of the aria-labels improved a little these controls for blind screen reader users, the labeling is still lss than ideal.

## Proposal

The visible text of the controls e.g. Status, Publish, Link, Author, etc. are just 'visual labels'. Plain text with no semantic association with the controls. I'd like to propose to make the visual labels the actual buttons to edit the settings and the displayed values just plain text. Quick screenshot to better illustrate:

![Screenshot 2024-07-11 at 16 28 06](https://github.com/user-attachments/assets/ea6a5b42-8316-4174-98b7-dc81d055ae17)

- The visual change would be minimal.
- The controls visible text would clearly identify the controls action.
- Some special cases would need some adjustments e.g. the 'Template Options' to avoid the need tor a tooltip.
- Although not 100% ideal, I'd would even consider to not use any aria-label attribute to clarify the controls as they would be sufficiently understandable.

Optionally:

Consider to make the whole 'row' the actual clickable area. A larger target size always helps. The visibel text of the value could be made hidden from assistive technology and used for an aria-describedby attribute to communicate the current value in an accessible way. Quick example:

![Screenshot 2024-07-11 at 16 27 34](https://github.com/user-attachments/assets/ef399fde-4f54-4d2d-8a2d-57fd0c6c18d4)

Pseudo markup:
```
<button aria-describedby="status-description" ... >
    Status
    <span aria-hidden="true">Draft</span>
</button>
<div id="status-description" hidden>Current status: Draft</div>
```

Cc @joedolson @WordPress/gutenberg-design 

### Step-by-step reproduction instructions

- Edit a post
- Observe the Post summary panel in the Inspector
- Inspect the source of the buttons and observe their text doesn't communicate the button action.
- Observe the buttons aria-label mismatches the visible text.

E.g.:
```
<button type="button" aria-label="Change date: July 22, 2024 1:40&nbsp;pm UTC+5:45">
    July 22, 2024 1:40 pm UTC+5:45
</button>
```

### Please confirm that you have searched existing issues in the repo.

- [X] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [X] Yes

**Comment by hanneslsm** [comment]
>  I'd like to propose to make the visual labels the actual buttons to edit the settings and the displayed values just plain text. Quick screenshot to better illustrate:

From UX perspective I'm not a big fan of this. 
Sighted users are used to click on the value they want to change, therefore it should be blue and indicating that it's clickable. Dropdowns or other elements use this UX pattern.

> Consider to make the whole 'row' the actual clickable area.

That'd be fine, if we keep the description / label blue.

**Comment by hanneslsm** [comment]
>  Plain text with no semantic association with the controls. I'd like to propose to make the visual labels the actual buttons to edit the settings and the displayed values just plain text.

The only way I could imagine this could work is by changing visually the values to make it clear "it's data". I'm not a fan of this either though.
Quick draft:
<img width="444" alt="image" src="https://github.com/user-attachments/assets/a9a80be1-3526-4870-a6cb-91ae252504be">

**Comment by afercia** [comment]
Thanks for your feedback.

> Sighted users are used to click on the value they want to change

Any data, user testing or any other reference to support this statement?

I'd like to remind once again, as discussed since https://github.com/WordPress/gutenberg/issues/470 more than _seven_ years ago, that this isn't just about visuals. It's about semantics, accessibility, and how to properly label _buttons_.

Using this kind of text for a button:

![352523885-e63d4e9e-8384-4f57-88f1-9916cc22c723](https://github.com/user-attachments/assets/504050ee-bfbd-4669-895c-81793041dbbe)

or this:

![Screenshot 2024-07-26 at 16 02 08](https://github.com/user-attachments/assets/8a1fc11e-12cf-4f20-8be6-253573f5b77c)

or this:

![Screenshot 2024-07-26 at 16 02 25](https://github.com/user-attachments/assets/c5a893c6-7611-4cd9-a3a0-00c7dec3a77b)

is, honestly, very arguable.

> Dropdowns or other elements use this UX pattern.

That's right and it's because a `<select>` element is designed to show the selected value. It's an entirely different control compared to a button.  That's also why `<select>` elements need a _visible label_ to communicate what they are about, because the selected value may not communicate it appropriately.

**Comment by richtabor** [comment]
> Sighted users are used to click on the value they want to change
> Any data, user testing or any other reference to support this statement?

Selecting "Link" to change the URL is objectively less clear than selecting the value you want to change. Like @hanneslsm mentioned, akin to dropdowns, but also the same UX for rows/tables of information. You don't click on labels to change a value—you interact with the field, the value.

I too disagree with making the labels the buttons. 

> I'd like to remind once again, as discussed since https://github.com/WordPress/gutenberg/issues/470 more than seven years ago, that this isn't just about visuals. It's about semantics, accessibility, and how to properly label buttons.

I appreciate the reminder regarding the significance of semantics and accessibility. However, it’s important to recognize that visuals play a crucial role in user experience. Visual design directly impacts usability, cognitive load, and overall user satisfaction. Omitting visual design is detrimental to the project; let’s aim for balance.

**Comment by jameskoster** [comment]
Making the entire row interactive is an interesting idea. Would it improve a11y if we did that, but didn't change the appearance?

**Comment by joedolson** [comment]
It seems like the only argument here is against switching the colors so that the label is blue and the value is non-interactive text; the more important change is making the entire region clickable, so that the label is part of the button, and it doesn't seem to me like anybody has a problem with that change. 

From my perspective, if the entire thing is clickable, I don't really care which part of it is blue, as long as a portion of the text contains a visual affordance that this is interactive. Users are already accustomed to the fact that you can click on a `label` element and it sets focus to the associated interactive element; so if the label is now clickable, that's great - it's an expected interaction. 

So, in response to @jameskoster - Yes, it would absolutely improve accessibility if we combine both texts into the button without changing their appearance. The important change is that the purpose of the control would now be included in the text without requiring a mismatch between an `aria-label` and the visible text. While the text of the button would still change, it would only be part of the control's name, not the entire name - and the constant part would be the first part of the text, which is good for accessibility. 

As far as I'm concerned, making this change with no visible alteration would be a significant accessibility improvement. 

I'll also second the removal of `aria-label` attributes in this panel; it's always preferable to have the visible text be the label for a control, rather than depending on an alternate string.

**Comment by afercia** [comment]
> Selecting "Link" to change the URL is objectively less clear than selecting the value you want to change

The problem I have with this kind of statements is that they're not supported by any data, research, specification, etc. Stating that something is 'objectively less clear' without providing arguments to backup that statament doesn't help in any way collaboration in this project. It doesn't help to improve the user experience too. 

> However, it’s important to recognize that visuals play a crucial role in user experience.

I totally agree. And, honestly, I would say the overall visual experience in the editor isn't great.

> Visual design directly impacts usability, cognitive load, and overall user satisfaction. Omitting visual design is detrimental to the project; let’s aim for balance.

To me there's no such a thing like 'visual design' that is separated by semantics, accessible design, universal design or whatever we want to call them. There is 'web design', which is a discipline that includes many aspects and none of them is separated from the other ones. WordPress aims to be as accessible as possible. As such, designers must work in the constraints of accessibiity requirements, which is not different from other constraints they are already working with. It's a requirement of the project, not something to be continuously dismissed, deprioritized, or moved to the level of 'remediation', when it's already too late.

> You don't click on labels to change a value—you interact with the field, the value.

Input fields and buttons are pretty different. These arebuttons, so the example above is out of context.

**Comment by jameskoster** [comment]
I reckon we should try making the entire row a button, potentially as a part of the [Data Forms](https://github.com/WordPress/gutenberg/issues/59745) work.

**Comment by afercia** [comment]
>> You don't click on labels to change a value—you interact with the field, the value.

> Input fields and buttons are pretty different. These arebuttons, so the example above is out of context.

I do realize my previous statement above needs to be expanded a bit. It's basically the same old issue from https://github.com/WordPress/gutenberg/issues/470

## Input fields
Sure, to change a value within an input field, we do click the value. The input field could be empty though, so there's nov value to 'click'. Also, clicking a proprly associated label moves focus to teh input thus achieving the same effect of clicking the value. But yes, _visually_ we click the value. 

Native input fields are simple, intuitive to use: Click the spot where the value is supposed to be set and edit it. 

![input](https://github.com/user-attachments/assets/877fad09-7367-4a8a-9387-f7e36e388edb)

However, the controls in the summary panel are not input fields.

_They are buttons that open the UI to edit the values_.

## Buttons

Native button elements are meant to submit a form or perform an action. We don't put a value or state inside the button. Instead, we do label the button to communicate what the button action is. Consider this example:

![button](https://github.com/user-attachments/assets/0be3f0f1-6296-4691-a5b4-53f6c10b6f93)

Would you ever put the Permalink value inside the button? That would be a very arguable usage of a button. Actually, that's what the summary panel does and that pattern is in place since seven years. The only proper usage for buttons is to label them to communicate the associated action. In the example above that is 'Edit'.

## What to do in the Sumary panel
Ideally, I'd love to see the buttons have a meaningful label to communicate the associated action. E.g.:
- Edit Status 
- Edit Publish date
- Edit Link
- etc.

As a compromise, I'd like to see the buttons be labeled _at least_ with the 'what' as in: instead of explicitly communicate the action 'Edit', using the name of the setting would be an improvement. For no reason these buttons should be labeled with the setting value. This way, the button labels would at least clearly communicate _waht_ they are about. E.g.:

- Status 
- Publish date
- Link
- etc.

Visually, the button could be the entire row. I'm confident that we can hide from assistive technology the visible value and communicate it via the accessible description (aria-describedby).

**Comment by afercia** [comment]
Looking into this, I think the [PostLastRevisionPanel](https://github.com/WordPress/gutenberg/blob/be5ce8aefa49b394a57aa540d9c9b091510ed8f1/packages/editor/src/components/post-last-revision/panel.js) is no longer used? If so, it could be removed.

**Comment by t-hamano** [comment]
> Looking into this, I think the [PostLastRevisionPanel](https://github.com/WordPress/gutenberg/blob/be5ce8aefa49b394a57aa540d9c9b091510ed8f1/packages/editor/src/components/post-last-revision/panel.js) is no longer used? If so, it could be removed.

Although this component is no longer used in Gutenberg, I don’t think it can be removed because it’s [a public component](https://github.com/WordPress/gutenberg/blob/39ed61df5b692fc54f3d119f8cb177ea9db47d6b/packages/editor/src/components/index.js#L54).

**Comment by afercia** [comment]
Got it. I guess it should be at least deprecated? And candidate for removal. If only there was a deprecation policy 😆
